packages/flutter/lib/src/cupertino/sliding_segmented_control.dart

```diff
@@ -0,0 +1,1037 @@
+const Radius _kThumbRadius = Radius.circular(6.93);
+// The amount of space by which to expand the thumb from the size of the currently
+// selected child.
+const EdgeInsets _kThumbInsets = EdgeInsets.symmetric(horizontal: 1);
+
+// Minimum height of the segmented control.
+const double _kMinSegmentedControlHeight = 28.0;
+
+const Color _kSeparatorColor = Color(0x4D8E8E93);
+
+const CupertinoDynamicColor _kThumbColor = CupertinoDynamicColor.withBrightness(
+  color: Color(0xFFFFFFFF),
+  darkColor: Color(0xFF636366),
+);
+
+// The amount of space by which to inset each separator.
+const EdgeInsets _kSeparatorInset = EdgeInsets.symmetric(vertical: 6);
+const double _kSeparatorWidth = 1;
+const Radius _kSeparatorRadius = Radius.circular(_kSeparatorWidth/2);
+
+// The minimum scale factor of the thumb, when being pressed on for a sufficient
+// amount of time.
+const double _kMinThumbScale = 0.95;
+
+// The minimum horizontal distance between the edges of the separator and the
+// closest child.
+const double _kSegmentMinPadding = 9.25;
+
+// The threshold value used in hasDraggedTooFar, for checking against the square
+// L2 distance from the location of the current drag pointer, to the closest
+// vertice of the CupertinoSlidingSegmentedControl's Rect.
+//
+// Both the mechanism and the value are speculated.
+const double _kTouchYDistanceThreshold = 50.0 * 50.0;
+
+// The corner radius of the segmented control.
+//
+// Inspected from iOS 13.2 simulator.
+const double _kCornerRadius = 8;
+
+// The spring animation used when the thumb changes its rect.
+final SpringSimulation _kThumbSpringAnimationSimulation = SpringSimulation(
+  const SpringDescription(mass: 1, stiffness: 503.551, damping: 44.8799),
+  0,
+  1,
+  0, // Everytime a new spring animation starts the previous animation stops.
+);
+
+const Duration _kSpringAnimationDuration = Duration(milliseconds: 412);
+
+const Duration _kOpacityAnimationDuration = Duration(milliseconds: 470);
+
+const Duration _kHighlightAnimationDuration = Duration(milliseconds: 200);
+
+class _FontWeightTween extends Tween<FontWeight> {
+  _FontWeightTween({ FontWeight begin, FontWeight end}) : super(begin: begin, end: end);
+
+  @override
+  FontWeight lerp(double t) => FontWeight.lerp(begin, end, t);
+}
+
+/// An iOS 13 style segmented control.
+///
+/// Displays the widgets provided in the [Map] of [children] in a horizontal list.
+/// Used to select between a number of mutually exclusive options. When one option
+/// in the segmented control is selected, the other options in the segmented
+/// control cease to be selected.
+///
+/// A segmented control can feature any [Widget] as one of the values in its
+/// [Map] of [children]. The type T is the type of the keys used
+/// to identify each widget and determine which widget is selected. As
+/// required by the [Map] class, keys must be of consistent types
+/// and must be comparable. The ordering of the keys will determine the order
```

Comment:
Actually I copied these from `CupertinoSegmentedControl` :sweat_smile: . The child widgets will be visually placed in the same order as they appear in the `Map`, if the `Map` is unordered then the children may appear in a different order every time the widget rebuilds.

Reply:
Maybe add a disclaimer like I see elsewhere: `The [children] argument must be an ordered [Map] such as a [LinkedHashMap]`.  I feel like it will be easier for people to end up with the order changing on them without understanding why.